src/analyses/cfg_dominators.h

```diff
@@ -115,6 +246,29 @@ class cfg_dominators_templatet
 protected:
   void initialise(P &program);
   void fixedpoint(P &program);
+
+  /// Goal of this is to process the tree in a post-order fashion.
+  /// Full explanation here: https://en.wikipedia.org/wiki/Tree_traversal.
+  ///
+  /// Post-ordering is when you pick one path of the tree to walk down to its
+  /// end, number that node, walk up one, do the same with the other path,
+  /// then repeat. When you've already numbered both children of a node, you
+  /// number it, then walk up one. This then continues until you've walked the
+  /// entire tree.
```

Comment:
This summary assumes that each node in the tree has two children. It also assumes a tree, rather than a graph. Maybe better to not give a summary, since tree traversals are such a fundamental idea in computer science that there are a gazillion explanations online.